base-theme/layouts/shortcodes/resource_link.html

```diff
@@ -2,7 +2,10 @@
 {{- $title := index .Params 1 | safeHTML -}}
 {{- $anchor_id := index .Params 2 | default ""}}
 {{- range where $.Site.Pages "Params.uid" $uuid -}}
-{{- if eq .Params.content_type "external-resource" -}}
+  {{- if eq .Params.content_type "external-resource" -}}
+    {{- if findRE "^(?:http|https|ftp|mailto)" $title -}}
+      {{- $title = (replace $title "/" "\\/") -}}
+    {{- end -}}
```

Comment:
This shortcode creates a new anchor tag whenever it sees a URL in the text.

So a markdown of
```html
<a href="https://google.com">https://google.com</a>
```

produces two anchor tags. 

```html
<a href="https://google.com"></a>
<a href="https://google.com">https://google.com</a>
```

I was unable to figure out why this is the case. My best guess is that the shortcode's output is run through an additional markdown renderer. 

Escaping the URL fixes the problem. Though a better solution might still exist.

Reply:
I took a look into this and I think I found an explanation. Although, I should preface this with the fact that I was able to take out revert `resource_link.html` to the old version without seeing the two anchor tags like you mentioned. Anyway, here's what I think is going on:

1. The content calls the shortcode with the `%` delimiters that tell Hugo that the shortcode will produce markdown that needs to be processed, i.e. `{{% resource_link "f24cc1d7-0985-4902-bf27-f9fc8a03e9b4" "Google" %}}`
2. The shortcode calls `{{- partial "external_resource_link" (merge .Params (dict "text" $title)) -}}`, which calls a partial and not a shortcode
3. The `external_resource_link` partial then gathers all the properties of the link and calls `{{- partial "link" (dict "href" $href "text" $text "class" $className "onClick" $onClick "target" "_blank") -}}`
4. The `link.html` partial builds and returns an HTML link

Hugo then goes to render the page, and since the original shortcode was called with `%` delimiters, it assumes that it is returning markdown. Therefore, the raw HTML link generated by `link.html` is processed as markdown, resulting in the behavior you saw. I think in general, it is problematic to have a shortcode that can generate markdown through one code path, and HTML through another. The `link.html` partial was created to be able to create links in templates that set more than just the `href` and inner text, rather than the limited syntax of markdown links which only let you set those properties. I can think of two solutions here:

1. Change the `else` block in `resource_link.html` to return a raw HTML link and change all calls to `resource_link` in content to use `<` / `>` delimiters, something like:
```
  {{- else -}}
    <a href="{{- .Permalink -}}{{- $anchor_id -}}">{{ $title }}</a>
  {{- end -}}
```
2. Create a separate shortcode, maybe `resource_link_external.html` and call that partial instead for external links but also with the `<` / `>` delimiters.

Reply:
Thanks for the explanation.

Reply:
@gumaerc, I looked at this again and you're exactly right about the cause.

I was hoping to find something that could mark the partial's output as "does not need markdown rendering" but I don't think such a feature exists in Hugo for `{{%` tags.

From the possible solutions you shared above, we might not be able to rely on 2. Because using a new shortcode might require us to create a new UI flow. 

I think 1 is our best option, however, it involves many steps. We'd have to:
- Update themes
- Update Studio (i.e. CKEditor syntax because `{{< resource_link...` is not currently supported)
- Update content and publish it

This issue with the URL as text is the only glitch I have found. I think we can apply this PR code as a temporary fix while we work on the proper solution. @gumaerc what do you think?

In any case, I'll write up an issue for this.